src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -98,7 +98,11 @@ public void setPersons(List<Person> persons) {
         internalList.setAll(persons);
     }
 
+    /**
+     * Sorts the address book's internal list according to the given comparator
+     */
     public void sortPersons(Comparator<Person> c) {
+        assert internalList.size() > 0;
```

Comment:
This internalList assertion seems slightly problematic. Given an invalid json save, the json save would be wiped out and replaced with an empty json save. This will create an internalList with size 0 which seems common enough to not warrant an assert. Perhaps a simple `assert c != null` would suffice.